tests/integration/Microsoft.Bot.Builder.Integration.AspNet.Core.Tests/ServiceCollectionExtensionsTests.cs

```diff
@@ -48,21 +50,7 @@ public void WithExplicitNullConfigurationCallback()
                 serviceCollectionMock.Verify(sc => sc.Add(It.Is<ServiceDescriptor>(sd => sd.ServiceType == typeof(IBot) && sd.ImplementationType == typeof(ServiceRegistrationTestBot) && sd.Lifetime == ServiceLifetime.Transient)));
                 serviceCollectionMock.Verify(sc => sc.Add(It.Is<ServiceDescriptor>(sd => sd.ServiceType == typeof(BotFrameworkAdapter) && sd.Lifetime == ServiceLifetime.Singleton)));
                 serviceCollectionMock.Verify(sc => sc.Add(It.Is<ServiceDescriptor>(sd => sd.ServiceType == typeof(IConfigureOptions<BotFrameworkOptions>))), Times.Never());
-            }
-
-            [Fact]
-            public void WithConfigurationCallback()
-            {
-                var serviceCollectionMock = new Mock<IServiceCollection>();
-
-                serviceCollectionMock.Object.AddBot<ServiceRegistrationTestBot>(options => 
-                {
-                    options.Should().NotBeNull();
```

Comment:
>options [](start = 20, length = 7)

This Action<> function was never called (the service is a mock!).  This is identical to previous test.

Reply:
So it's true that wasn't called and so the assertion in there is not useful, however this test is not identical to the previous tests. As the names suggest, one is calling with an explicit null configuration callback and the other is non-null. These hit different code paths and thus, it's testing different, specific behaviors. Instead of the `options.Should().NotBeNull();` check in the config action what this test should have really added that the other test didn't have was a call to verify that the `serviceCollectionMock`'s  `Configure(...)` method was called with the expected `Action<BotFrameworkOptions>` instance. I would be happy to fix this test up if what I'm saying isn't clear.

Reply:
@drub0y : Would love the support, as I couldn't figure out how to fix the test and would love to see how to do it. Moq with extension methods are not super friendly together.

Reply:
Sure, I'll push up the changes I was talking about.